app/docs/md/conventions/components.md

```diff
@@ -0,0 +1,253 @@
+```
+
+The client-side code will look for this attribute and only run if your component hasn’t already been "enhanced". Thus avoiding an unnecessary render pass.
+
+If you are have an existing Enhance Element you can always import it into your Component and use it as your `render` function.
+
+<doc-callout level="none" mark="✨">
+
+**[Learn about the HTML render function](/docs/elements/html)**
+
+</doc-callout>
+
+## Rehydration
+Enhance Components handle rehydration by listening to attribute changes. Any change to an attribute listed in the `observedAttributes` will trigger a `<attribute name>Changed` method. For example if you are observing the `title` attribute of our `my-card` component any time that attribute value is updated the `titleChanged` method will be executed. This enables you to write surgical DOM updates which will always be the most performant way to update your page.
+
+## DOM Diffing
+Other frameworks supply a DOM diffing solution and Enhance Components are no different but we believe it should be on an opt-in basis. To enabled DOM diffing in our `my-card` component we will add the `MorphdomMixin` class from `@enhance/morphdom-mixin`.
+
+<doc-code filename="app/components/my-card.mjs">
+
+```javascript
+import CustomElement from '@enhance/custom-element'
+import MorphdomMixin from '@enhance/morphdom-mixin'
+
+export default class MyCard extends MorphdomMixin(CustomElement) {
+  render({ html, state }) {
+    const { attrs={} } = state
+    const { title='default' } = attrs
+    return html`
+      <style>
+        :host {
+            position: relative;
+            display: flex;
+            flex-direction: column;
+            min-width: 0;
+            word-wrap: break-word;
+            color: black;
+            background-color: #fff;
+            background-clip: border-box;
+            border: 1px solid rgba(0,0,0,.125);
+            border-radius: 0.25rem;
+        }
+        .card-img {
+            width: 100%;
+            border-top-left-radius: calc(0.25rem - 1px);
+            border-top-right-radius: calc(0.25rem - 1px);
+        }
+        .card-body {
+            flex: 1 1 auto;
+            padding: 1.25rem;
+        }
+        .card-title {
+            margin-bottom: 0.75rem;
+            font-size: 1.25rem;
+            font-weight: 500;
+        }
+      </style>
+      <slot name="image"></slot>
+      <div class="card-body font-sans">
+        <h5 class="card-title">${title}</h5>
+        <slot></slot>
+      </div>
+    `
+  }
+
+  static get observedAttributes() {
+    return [ 'title' ]
+  }
+}
+
+customElements.define('my-card', MyCard)
+```
+
+</doc-code>
+
+Once added the `MorphdomMixin` will handle updating the DOM whenever an `observedAttributes` is modified. The `<attribute name>Changed` methods are no longer necessary. Instead on an attribute change the `render` method will be re-run and the output will be compared against the current DOM. Only the modified DOM nodes will be updated.
```

Comment:
I will work on a Morphdom doc that should be linked to here in a callout.

Reply:
We could optionally make a mention about how Morphdom does a String based diff on an the actual HTML element and not a virtual DOM diff so every element you want diffed needs to have a string change or unique id.

Reply:
Wouldn't hurt to put it _something_ in the paragraph as well